solutions/gold/usaco-696.mdx

```diff
@@ -99,8 +99,81 @@ public class PromotionCounting {
 }
 ```
 </JavaSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+const int mx = 1e5 + 1;
+int n, t = 0, out[mx], in[mx], bit[mx], ans[mx];
+vector<int> adj[mx];
```

Comment:
just...use vectors...

Reply:
only using vectors for bit and ans for convenience; i'd rather not have to add out, in, adj as arguments for dfs

Reply:
what's wrong with declaring vectors globally

Reply:
also uh put the declarations on separate lines

Reply:
either i'd have to resize it within main() (more imo unnecessary code) or i set it with size mx in the global space anyway(pointless?)

Reply:
nvm